src/components/pages/Camera/VideoPreview.tsx

```diff
@@ -0,0 +1,196 @@
+// warning. https://github.com/facebook/react-native/issues/12981
+console.ignoredYellowBox = ["Setting a timer"];
+
+import * as React from "react";
+import firebase from "firebase";
+import { View, Text, StyleSheet, Button } from "react-native";
+import Video from "react-native-video";
+
+const BYTES_PER_MIB = 1024 * 1024;
+const MAX_MB = 60;
+const MAX_VIDEO_SIZE = MAX_MB * BYTES_PER_MIB;
+
+type VideoPreviewProps = {
+  videoUri: string;
+  videoUploadRef: firebase.storage.Reference;
+  onVideoUploaded: (videoDownloadUrl: string) => any;
+  onRetakeClicked: () => any;
+};
+
+type VideoStateProps = {
+  errorMessage?: string;
+  uploadStatus: UploadStatus;
+  uploadedBytes: number;
+  totalBytes: number;
+  videoDownloadUrl?: string;
+};
+
+enum UploadStatus {
+  NOT_STARTED,
+  UPLOADING,
+  UPLOADED
+}
+
+export class VideoPreview extends React.Component<
+  VideoPreviewProps,
+  VideoStateProps
+> {
+  constructor(props: VideoPreviewProps) {
+    super(props);
+    this.state = {
+      uploadStatus: UploadStatus.NOT_STARTED,
+      uploadedBytes: 0,
+      totalBytes: 0
+    };
+  }
+
+  uploadVideo = async (videoUploadRef: firebase.storage.Reference) => {
+    const response = await fetch(this.props.videoUri);
+    const blob = await response.blob();
+    //@ts-ignore Blob does not have data type
+    if (blob.data.size > MAX_VIDEO_SIZE) {
+      this.setState({
+        errorMessage:
+          "The video size is larger than " +
+          MAX_VIDEO_SIZE +
+          "MB. Please retake your video."
+      });
+      return;
+    }
+
+    // TODO: Transcode video to make it smaller.
+    const metadata = {
+      //@ts-ignore Expo Blob does not have data type
+      contentType: blob.data.type
+    };
+    const uploadTask = videoUploadRef.put(blob, metadata);
+    uploadTask.on(
+      firebase.storage.TaskEvent.STATE_CHANGED,
+      (s: any) => {
+        const snapshot = s as firebase.storage.UploadTaskSnapshot;
+        this.setState({
+          uploadStatus: UploadStatus.UPLOADING,
+          uploadedBytes: snapshot.bytesTransferred,
+          totalBytes: snapshot.totalBytes
+        });
+      },
+      err => {
+        this.setState({
+          errorMessage: "Could not upload. Please try again.",
+          uploadStatus: UploadStatus.NOT_STARTED
+        });
+      },
+      async () => {
+        const videoDownloadUrl = await uploadTask.snapshot.ref.getDownloadURL();
+        if (videoDownloadUrl) {
+          this.setState({ uploadStatus: UploadStatus.UPLOADED });
+          this.props.onVideoUploaded(videoDownloadUrl);
+        } else {
+          this.setState({
+            errorMessage: "Could not retrieve download URL. Please try again.",
+            uploadStatus: UploadStatus.NOT_STARTED
+          });
+        }
+      }
+    );
+  };
+
+  componentWillMount() {
+    // Validate video state.
+    if (!this.props.videoUri) {
+      console.warn(
```

Comment:
Do we want the `console.warn` ?

Reply:
yup, for now, I will have a PR for drop-down style errors